### PR TITLE
[Merged by Bors] - chore(Combinatorics/SimpleGraph): Generalize few lemmas in `AdjMatrix` to `NonAssocSemiring α`

### PR DESCRIPTION
**Pull request by Rida-Hamadani**
Generalize some lemmas from `ℕ`, `Ring α`, and `Semiring α`, to `NonAssocSemiring α`.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Combinatorics/SimpleGraph/AdjMatrix.lean [hunk 1]
```diff
@@ -278,21 +278,20 @@ theorem adjMatrix_pow_apply_eq_card_walk [DecidableEq V] [Semiring α] (n : ℕ)
       simp at hxy
 #align simple_graph.adj_matrix_pow_apply_eq_card_walk SimpleGraph.adjMatrix_pow_apply_eq_card_walk
 
-/-- The sum of the identity, `G.adjMatrix ℕ` and `(G.adjMatrix ℕ).compl` is the all-ones matrix. -/
-theorem one_add_adjMatrix_add_compl_adjMatrix_eq_allOnes [DecidableEq V] :
-    1 + G.adjMatrix ℕ + (G.adjMatrix ℕ).compl = Matrix.of fun _ _ ↦ 1 := by
+/-- The sum of the identity, the adjacency matrix, and its complement is the all-ones matrix. -/
```
Comment (riccardobrasca) [inline Mathlib/Combinatorics/SimpleGraph/AdjMatrix.lean:281]: There is still the identity in the formula, right?

Comment (Rida-Hamadani) [reply Mathlib/Combinatorics/SimpleGraph/AdjMatrix.lean:281]: Yes

Comment (riccardobrasca) [reply Mathlib/Combinatorics/SimpleGraph/AdjMatrix.lean:281]: Would you mind fix the doc? Thanks!

Comment (Rida-Hamadani) [reply Mathlib/Combinatorics/SimpleGraph/AdjMatrix.lean:281]: The docs start by mentioning the identity, is there anything wrong with it?

Comment (riccardobrasca) [reply Mathlib/Combinatorics/SimpleGraph/AdjMatrix.lean:281]: Sorry, I somehow misread it.

**Comment by riccardobrasca** [comment]
Thanks!

bors merge

**Pull request closed without merging**